event/src/unvalidated/event.rs

```diff
@@ -65,8 +64,7 @@ where
             let block_bytes = car_blocks
                 .get(cid)
                 .ok_or_else(|| anyhow!("Time Event CAR data missing block for path index"))?;
-            blocks_in_path.push(block);
```

Comment:
this was the bug.  We pushed the block before we updated it, meaning we pushed the first block twice and dropped the last block

Reply:
😨